include/EASTL/vector.h

```diff
@@ -890,7 +890,19 @@ namespace eastl
 			else if(n < (size_type)(mpEnd - mpBegin))
 				resize(n);
 
+		#if EASTL_MOVE_SEMANTICS_ENABLED
+			this_type     temp;
+			size_type     nNewSize = size();
+			pointer const pNewData = DoAllocate(nNewSize);
+
+			pointer pNewEnd = eastl::uninitialized_move_ptr(mpBegin, mpEnd, pNewData);
+
+			temp.mpBegin    = pNewData;
+			temp.mpEnd      = pNewEnd;
+			temp.mpCapacity = pNewEnd;
+		#else
 			this_type temp(*this);  // This is the simplest way to accomplish this, 
```

Comment:
Take a look at the 'shrink_to_fit' implementation.  It utilizes the vector constructor which takes a pair of iterators (begin, end) and passes move iterators to that constructor.

		#if EASTL_MOVE_SEMANTICS_ENABLED
			this_type temp = this_type(move_iterator<iterator>(begin()), move_iterator<iterator>(end()), mAllocator);